### PR TITLE
space dragon qol, refactoring and misc

### LINKED ISSUE #18871: Carp portal stuck and wont finish charging
## Test Merges: 
Not Supplied / None
## Reproduction:
Carp portal from space dragon stays at 99.7917%

 Submitted by: PibeCantina (341102461504716800)

### LINKED ISSUE #18921: Space dragon deletion thing thats really stupid and dumb plz fix pls pls pls pls
### Severity

Inconvenience

### Short Description

Space dragon's victims are punished for when the space dragon loses.

### Reproduction

1. get vored by space dragon
2. space dragon fails to protect its rifts and ceases to exist
3. haha fuck you your also deleted.

### Round ID

47219

### Test Merges

_No response_

### BYOND Username

fireclaw787

### Was this done on a locally hosted or non-Yogstation hosted server?

No

### Additional information

I think it would be better if the dragon simply gibbed you like it used to.

### PR DESCRIPTION
**Pull request by tattax**
ports what was leftover in the original pr in respect to refactoring, QOLs and etc, together with other various bugfixes
since the original one felt like something was missing from the antag (it was)

mostly from : https://github.com/tgstation/tgstation/pull/68988

closes: https://github.com/yogstation13/Yogstation/issues/18871
closes: https://github.com/yogstation13/Yogstation/issues/18921
raises weight from 5 to 7, since 5 is really low considering space dragon doesn't immediately ends round like blob (weight 8)

### gust is now activated with _ALT-CLICK_, instead of being an action

dragon roar SFX: 

https://vocaroo.com/1cRq1jBzGDD1

dragon with color (name is also customizable):
image

:cl: IndieanaJones, Timberpoes, tatax
tweak: space dragon qol: allows you to choose color and name, adds nightvision, increases dragon weight from 5 to 7
soundadd: space dragon roar
/:cl:

### CODE REVIEW

**Comment by MajManatee** [comment]
I really dont like how the colouring is done.

**Comment by cuackles** [comment]
can you show more of the colored drakes stuff like some dark colors or very light colors

**Comment by tattax** [comment]
image

**Comment by MajManatee** [comment]
Might have to be limited in colours to prevent retard mcgee's from making the ugliest space dragon possible

**Comment by tattax** [comment]
read the code mr maintainer

File: code/modules/antagonists/space_dragon/carp_rift.dm [hunk 1]
```diff
@@ -0,0 +1,242 @@
...
+/// The carp rift is currently charging and has output a final warning.
+#define CHARGE_FINALWARNING 1
+/// The carp rift is now fully charged.
+#define CHARGE_COMPLETED 2
+
+/datum/action/innate/summon_rift
+	name = "Summon Rift"
+	desc = "Summon a rift to bring forth a horde of space carp."
+	background_icon_state = "bg_default"
+	icon_icon = 'icons/mob/actions/actions_space_dragon.dmi'
```
Comment (tattax) [inline code/modules/antagonists/space_dragon/carp_rift.dm]:
```suggestion
	button_icon = 'icons/mob/actions/actions_space_dragon.dmi'
```

**Review by ToasterBiome — APPROVED** [review]

**Pull request merged**